metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/SearchController.java

```diff
@@ -45,4 +49,18 @@
   ResponseEntity<SearchResponse> search(final @ApiParam(name = "searchRequest", value = "Search request", required = true) @RequestBody SearchRequest searchRequest) throws RestException {
     return new ResponseEntity<>(searchService.search(searchRequest), HttpStatus.OK);
   }
+
```

Comment:
It's a legitimate concern but I think synchronous is ok here because Elasticsearch is intended to provide results in real-time.  The http javascript APIs we use are asynchronous so it shouldn't make a difference in the UI.  If this were a batch operation (Hive, Map Reduce) I would agree that it should be an async call.